### ISSUE TITLE
Llama prefill incorrect values

### ISSUE THREAD

**Issue by mvasiljevicTT** [tenstorrent/tt-mlir#8068]
Prefill produces zero logits for some users, same for any number of layers (0, 5, 10, 13, 18, 27 and 30 for llama galaxy on UF-EV-A5-GWH01).

When model is run in `--decode-only` mode it works fine for all users.

This was not present on [this mlir commit](https://github.com/tenstorrent/tt-mlir/commit/a53ca15ef0d7e17ca7ecbac52698302539bfaa71) used as base for CIs on [this PR](https://github.com/tenstorrent/tt-mlir/pull/8028) where all values were ok in all models, so the issue can be bisected from it.

**Comment by mvasiljevicTT** [comment]
[This CI](https://github.com/tenstorrent/tt-xla/actions/runs/24697357671/job/72320741982) was run on tt-xla that uses [this mlir commit](https://github.com/tenstorrent/tt-mlir/commit/026940fee6f2343b2f0516fce5a2efc7562b26a9) and might already show signs of this issue.

**Comment by mvasiljevicTT** [comment]
It seems that nightly from 21.4. already had that issue, while 20.4. ~~did not~~ **did not have complete garbage** (still had some difference in users), which is when mlir uplifts were done. 

Uplifts on that day jumped 40 mlir commits, including 2 metal uplifts, which are most probable root cause. 

Metal should be bisected from 3fa4d753550dba1d4aacc9af45b111ae540f63fc to 90c914ef258b5cc92ad172f3604b784ec77253ca checking where user 3 in llama_3_2_1b started producing complete garbage, like in [run from 21.4](https://github.com/tenstorrent/tt-xla/actions/runs/24697252081/job/72278738305), compared to a bit less garbage output like on [20.4.](https://github.com/tenstorrent/tt-xla/actions/runs/24642681050/job/72084379470) that was solved with decode sdpa fix

**Comment by sdjordjevicTT** [comment]
Is this related to this?
https://github.com/tenstorrent/tt-mlir/issues/8042

**Comment by mvasiljevicTT** [comment]
@sdjordjevicTT might be, but #8042 seems to be decode only issue, while this is prefill only

**Comment by mvasiljevicTT** [comment]
For [this run](https://github.com/tenstorrent/tt-xla/actions/runs/24781748566/job/72519554720) I experienced that issue locally on prefill while on CI it works fine, so it might only seem to be prefill only while it is just random 🤨

**Comment by mvasiljevicTT** [comment]
It seems that in [tt-metal llama demo](https://github.com/tenstorrent/tt-metal/blob/main/models/demos/llama3_70b_galaxy/tests/test_llama_accuracy.py#L357) only first user is checked, so there is possibly metal issue with some of the ops not working correctly for whole batch

**Comment by mvasiljevicTT** [comment]
Might be related: https://github.com/tenstorrent/tt-metal/issues/35509, though there all outputs are ok just different. In our case, when output is different it is garbage

**Comment by mvasiljevicTT** [comment]
tracked further in [this issue](https://github.com/tenstorrent/tt-xla/issues/4785)
